src/apps/concepts/components/AddConceptsIcon.tsx

```diff
@@ -0,0 +1,208 @@
+import React from "react";
+import {Link} from "react-router-dom";
+import {
+    createStyles,
+    Fab,
+    makeStyles,
+    Menu,
+    MenuItem,
+    Theme,
+    Tooltip
+} from "@material-ui/core";
+import {
+    CONCEPT_CLASSES,
+    PREFERRED_SOURCES_VIEW_ONLY,
+    useAnchor
+} from "../../../utils";
+import {Add as AddIcon} from "@material-ui/icons";
+
+interface Props {
+    canModifyDictionary: boolean;
+    canModifySource: boolean;
+    containerUrl: string;
+    gimmeAUrl: Function;
+    linkedSource?: string;
+    preferredSource: string;
+}
+
+const useStyles = makeStyles((theme: Theme) =>
+    createStyles({
+        link: {
+            textDecoration: "none",
+            color: "inherit",
+            width: "100%"
+        },
+        largerTooltip: {
+            fontSize: "larger"
+        }
+    })
+);
+
+const AddConceptsIcon: React.FC<Props> = ({
+                                              canModifyDictionary,
+                                              canModifySource,
+                                              containerUrl,
+                                              gimmeAUrl,
+                                              linkedSource,
+                                              preferredSource
+                                          }) => {
+    const classes = useStyles();
+
+    const [
+        addNewAnchor,
+        handleAddNewClick,
+        handleAddNewClose
+    ] = useAnchor();
+    const [
+        customAnchor,
+        handleCustomClick,
+        handleCustomClose
+    ] = useAnchor();
+    const [
+        importExistingAnchor,
+        handleImportExistingClick,
+        handleImportExistingClose
+    ] = useAnchor();
+
+    const getAddConceptIcon = () => {
+        return canModifyDictionary ?
+             (
+                <Tooltip title="Add concepts" data-testid="addConceptsIcon">
+                    <Fab onClick={handleAddNewClick} color="primary" className="fab">
+                        <AddIcon/>
+                    </Fab>
+                </Tooltip>
+            ) :
+             (
+                <Tooltip title="Create custom concept" data-testid="addCustomConceptIcon">
+                    <Fab onClick={e => {
+                        handleCustomClick(e);
+                        handleAddNewClose();
+                    }} color="primary" className="fab">
+                        <AddIcon/>
+                    </Fab>
+                </Tooltip>
+            )
+    };
+
+    const getMenuForAddConceptIcon = () => {
+        return canModifyDictionary ?
```

Comment:
Can we break this function into even smaller function. And use if else instead of ternary operator for better readability

Reply:
@rakesh-1010 It has one <Menu> which is not required to be divided into smaller functions. Let me know if you have other thoughts.